--- a/meson.build
+++ b/meson.build
@@ -548,8 +548,7 @@ config_h.set10('WITH_TEAMDCTL', enable_teamdctl)
 # polkit
 enable_polkit = get_option('polkit')
 if enable_polkit
-  # FIXME: policydir should be relative to `datadir`, not `prefix`. Fixed in https://gitlab.freedesktop.org/polkit/polkit/merge_requests/2
-  polkit_policydir = dependency('polkit-gobject-1').get_variable(pkgconfig: 'policydir', pkgconfig_define: ['prefix', nm_prefix])
+  polkit_policydir = join_paths(nm_datadir, 'polkit-1', 'actions')
   polkit_rulesdir = join_paths(fs.parent(polkit_policydir), 'rules.d')
 endif
 
@@ -609,8 +608,8 @@ if dbus_conf_dir == ''
   dbus_conf_dir = join_paths(dbus_dep.get_variable(pkgconfig: 'datarootdir', pkgconfig_define: ['prefix', nm_prefix]), 'dbus-1', 'system.d')
 endif
 
-dbus_interfaces_dir = dbus_dep.get_variable(pkgconfig: 'interfaces_dir',  pkgconfig_define: ['datadir', nm_datadir])
-dbus_system_bus_services_dir = dbus_dep.get_variable(pkgconfig: 'system_bus_services_dir', pkgconfig_define: ['datadir', nm_datadir])
+dbus_interfaces_dir = join_paths(nm_datadir, 'dbus-1', 'interfaces')
+dbus_system_bus_services_dir = join_paths(nm_datadir, 'dbus-1', 'system-services')
 
 enable_firewalld_zone = get_option('firewalld_zone')
 config_h.set10('WITH_FIREWALLD_ZONE', enable_firewalld_zone)
